tools/plugin_sync/plugin_sync.pt

```diff
@@ -0,0 +1,200 @@
+name "Plugin Synchronization"
+rs_pt_ver 20180301
+type "policy"
+short_description "A policy to automate publishing plugins to the catalog. See the [README](https://github.com/flexera/flexera-plugins/tree/master/tools/plugin_sync/) and [docs.rightscale.com/policies](http://docs.rightscale.com/policies/) to learn more."
+long_description ""
+severity "low"
+category "Operational"
+
+info( version: "1.0", publish: "false")
+
+parameter "param_email" do
+  type "list"
+  label "Email addresses"
+  description "Email addresses of the recipients you wish to notify"
+  min_length 1
+end
+
+parameter "force_upgrade" do
+  type "number"
+  label "Force Upgrade"
+  description "Force Upgrade, 0:False, 1:True"
+  default 0
+  allowed_values 0,1
+end
+
+auth "auth_rs", type: "rightscale"
+
+# get the active plugins from github
+datasource "ds_active_plugin_list" do
+  request do
+    verb "GET"
+    host "s3.amazonaws.com"
+    path "/rs-policysync-tool/active-plugin-list.json"
+  end
+  result do
+    collect jmes_path(response,"plugins") do
+      field "name", jmes_path(col_item, "name")
+      field "file_name", jmes_path(col_item, "file_name")
+      field "version", jmes_path(col_item, "version")
+      field "change_log", jmes_path(col_item,"change_log")
+    end
+  end
+end
+
+#get the list of published plugins
+datasource "ds_published_plugin_list" do
+  request do
+    auth $auth_rs
+    verb "GET"
+    host 'cloud-3.rightscale.com'
+    path join(["/cloud/v1/orgs/",rs_org_id,"/plugins"])
+    header "Api-Version", "1.0"
+    #query "view", "extended"
+  end
+  result do
+    collect jmes_path(response,"items") do
+      field "name",     jmes_path(col_item, "name")
+      field "file_name", jmes_path(col_item, "fileName")
+      field "info",     jmes_path(col_item, "info")
+      field "version",  jmes_path(col_item, "version")
+    end
+  end
+end
+
+datasource "ds_combined_plugin_list" do
+  run_script $js_combine_plugin_lists, $ds_active_plugin_list, $ds_published_plugin_list, $force_upgrade
+end
+
+script "js_combine_plugin_lists", type: "javascript" do
+  parameters "active_plugin_list", "published_plugin_list", "force_upgrade"
+  result "plugin_list"
+  code <<-EOS
+  var plugin_list = [];
+  var old_plugin_list = {};
+  var github_raw_url = 'https://raw.githubusercontent.com/flexera/flexera-plugins/master/';
+
+  // create a map of published templates
+  for (var index = 0; index < published_plugin_list.length; index++) {
+    var published_plugin = published_plugin_list[index];
+    old_plugin_list[published_plugin.name] = published_plugin;
+  }
+
+  // loop through active plugins from s3 json file
+  for ( var i = 0; i < active_plugin_list.length; i++) {
+    var active_plugin = active_plugin_list[i]
+    // map catalog plugin with json list
+    var account_plugin = old_plugin_list[active_plugin.name]
+
+    var plugin_name = ""
+    var account_version = "0.0.0"
```

Comment:
I am guessing that account_version is the version of plugin in the account ? Am I correct ?

Reply:
yes, that is the version that is in the catalog.